### ISSUE TITLE
Is 6.5 million sha256 virus hashes enough?

### ISSUE THREAD

**Issue by Piggo4676** [FossAV/FossAV#18]
Just checking, are the 6.5 million virus hashes that I added enough?

**Comment by UniqueName12345** [comment]
Good enough.

**Comment by PPPDUD** [comment]
Yes, but please remember that there's this weird quirk in hashlib that
means you'll need to get the hashes from Python or else they won't match.

**Comment by Piggo4676** [comment]
wait really? thats dumb. Any way to get around that?

**Comment by PPPDUD** [comment]
No. You'll have to collect those 6.5 viruses in a folder and then hash them
in Python. I know, it's not very convenient.

**Comment by Piggo4676** [comment]
Why does python do this? Do you know?

**Comment by Piggo4676** [comment]
> No. You'll have to collect those 6.5 viruses in a folder and then hash them in Python. I know, it's not very convenient.

Actually, I tested, and the hashes seem to be the same. They should be the same if they use the same encoding...

**Comment by PPPDUD** [comment]
Weird. It never worked for me.

**Comment by Piggo4676** [comment]
hmmm, that's strange. idk why.

**Comment by PPPDUD** [comment]
I was using Python 3.11 on Windows 11, if you were wondering.

**Comment by Piggo4676** [comment]
same, i also use w11... I tested using the browser and finding a random files hash using just a site and then I compared it to the hash that python generated for me and they where the same...

**Comment by PPPDUD** [comment]
Huh. Even weirder.

**Comment by Siradankullanici** [comment]
I have almost 45 million examples. Please give me hash.

**Comment by PPPDUD** [comment]
Are you asking for one of us to hash your live virus samples? I might be
interested.

**Comment by Piggo4676** [comment]
what do you mean by that? Do you mean almost 45 million hashes or 45 million viruses?

**Comment by Siradankullanici** [comment]
45 million hashes.

**Comment by Piggo4676** [comment]
holy guack... No chance you could give them to me? (plz) If yes, could you give me a download link?

**Comment by Siradankullanici** [comment]
It's not update to date but here you go: https://mega.nz/file/dfBk2bZL#Q9a2wMZrWzsJ-dclQTmGBDQ8w--vIvgtUWwlEH8Psmg

**Comment by Piggo4676** [comment]
Thank you!!!!!!!!!!!!!!!!!

**Comment by Piggo4676** [comment]
are most of the hashes md5? or are  most of the hashes duplicates? Also, did you get them from a source or multitple (I already use the full_sha256 source)? If multiple, can you give the source to me? if you find viruses yourself, can you get the sha256 versions of the viruses? or is it md5 so that it takes less space? also thanks btw for the hashes.

**Comment by Siradankullanici** [comment]
Most of them from virusshare. Some of them are from another sources (Example: https://www.youtube.com/watch?v=hEY7p_QLXN8&list=PLXRw3U-U9Y6LV73_nwSV4bB9hfARne6Yf  https://www.youtube.com/watch?v=qhbp5JjpRp8&list=PLkzUkNupwT3S-lwWGfG6D5Xs2-hKuxXT2) and from me. I can't convert to SHA256. You need to login virusshare for that. They have an SHA256 hashes.

**Comment by Piggo4676** [comment]
Is it free to get the sha256 hashes?

**Comment by PPPDUD** [comment]
Can someone sign up without using a private communications service such as
emailing? I'd prefer to sign up online.

**Comment by Siradankullanici** [comment]
https://virussign.com/ you can try this.

**Comment by Siradankullanici** [comment]
https://a4lg.com/downloads/vxshare/index.en.html also this.

**Comment by PPPDUD** [comment]
That website seems sketchy. If they wanted to save bandwidth, they wouldn't
allow free accounts at all. They also want you to email them, which they
can use to phish you.

**Comment by Siradankullanici** [comment]
I registered today to website easily.

**Comment by Piggo4676** [comment]
> I registered today to website easily.

"Load the Spam cannons. Prepare the Phishing bombs. Launch the Scam Nuke!!!"

**Comment by Piggo4676** [comment]
> You need to be login virusshare website. You can be register by mailing. It's free. Do you want join my Turko Antivirus project? I write in C#.

Maybe a Collab or something like that? (if that makes sense)

**Comment by Piggo4676** [comment]
Is there anything u need help on right now?

**Comment by Siradankullanici** [comment]
I need an real time protection like this: https://github.com/anic17/Batch-Antivirus my discord : Tulpar#8756

**Comment by Siradankullanici** [comment]
https://mega.nz/file/jp4XFCCJ#pyfolbnfEKwL-keDAxvfV9AwTKz09hVgl14YeVHmpnk https://mega.nz/file/Ww5wXCzB#175pO_eG2MPM1HqTv_PsmbVNrB9nTm_FNThEZY_uyMI my antivirus.
